### ISSUE TITLE
Ability to configure PUID and PGID in container

### ISSUE THREAD

**Issue by devedse** [revenz/FileFlows#365]
Hello, currently it's not possible to configure the PUID and PGID in the container. This results the user with PUID: 0 and PGID: 0 being used for all reads/writes on volume mounts.

**Explanation**

This means that if the host doesn't have a user with PUID: 0 and PGID: 0 with permissions on those folders the container throws errors that it can't access the files.

The other way around when the container writes files / folders on disk they also have the owner set to the user with PUID: 0 and PGID: 0. Which can result in the host user not being able to access the files easily.

**Fixing it**

A fix for this has been made in for example the Plex container (a user named abc is made):
https://github.com/linuxserver/docker-baseimage-ubuntu/blob/bionic/Dockerfile#L96
And the PUID and PGID are modified during start:
https://github.com/linuxserver/docker-baseimage-ubuntu/blob/bionic/root/etc/cont-init.d/10-adduser

The interesting thing about PUID and PGID is that this is the only mechanism determining file permissions. A user with username `devedse` could exist on the host with PUID 1000 and PGID 1000, whereas on the container this user could be called `abc`. As long as the PUID and PGID are the same all file reads/writes work.

**More background information**

https://docs.linuxserver.io/general/understanding-puid-and-pgid

https://vsupalov.com/docker-shared-permissions/

**Comment by saitoh183** [comment]
Im using the same container with this docker command 

```
docker create \ 
--name=FileFlows \
--network=saitohsmedia \
--restart=always \
-p 5000:5000 \
-e "TempPathHost=/data/docker_volumes/downloads/temp/fileflows/tempCont" \
-e "TZ=America/Toronto" \
-e PUID=1000 \
-e PGID=1000 \
-v /var/run/docker.sock:/var/run/docker.sock:ro \
-v FileFlows_config:/app/Data \
-v FileFlows_logs:/app/Logs \
-v FileFlows_temp:/temp \
-v docker_volumes:/data \
revenz/fileflows
```
and if i navigate to the container file system, the permissions seem to apply (In Bold)


root@bc9cbeb2ddb4:/app# ls -al
total 64
drwxr-xr-x 1 root root 4096 Nov 16 18:00 .
drwxr-xr-x 1 root root 4096 Nov 14 23:09 ..
**drwxr-xr-x 6 1000 abc  4096 Nov 14 23:09 Data**
-rwxr-xr-x 1 root root  980 Nov  4 17:58 docker-entrypoint.sh
drwxr-xr-x 1 root root 4096 Nov 14 23:16 FlowRunner
**drwxr-xr-x 3 1000 abc  4096 Nov 19 17:05 Logs**
drwxr-xr-x 3 root root 4096 Nov  4 17:57 Node
drwxr-xr-x 2 root root 4096 Nov  4 17:57 Plugins
-rw-r--r-- 1 root root   39 Nov  4 04:47 run-node.bat
-rw-r--r-- 1 root root   84 Nov  4 04:47 run-node.sh
-rw-r--r-- 1 root root   43 Nov  4 04:47 run-server.bat
-rw-r--r-- 1 root root   88 Nov  4 04:47 run-server.sh
drwxr-xr-x 3 root root 4096 Nov 16 18:00 Scripts
drwxr-xr-x 1 root root 4096 Nov 14 23:09 Server

**Comment by devedse** [comment]
@saitoh183 , that's interesting. I kinda see the same behaviour except that it seems to only apply to the folder itself, not the files inside it:

![image](https://user-images.githubusercontent.com/2350015/202878672-93bddd71-66eb-4ab1-918d-a43e902d2730.png)

Whereas my plex instance does have the user account set to all files:
![image](https://user-images.githubusercontent.com/2350015/202878727-70e66060-c259-43a1-8982-3bdf8537ddfe.png)

**Comment by saitoh183** [comment]
@devedse I didnt show you the right folders...That place is the same as yours since it is the container folders, they are all root like you...what did take the permissions was my other Volumes that are not par of the container. So yea @revenz the container might need fixing for the PUID and PGID to be passed if used. I also map a volume called data at the root of the container which is the data folder on my server where i do i my stuff. That one inherits all permissions. @devedse sorry for missing that!

**Comment by devedse** [comment]
It seems Synology has it's permissions configured a bit different (stricter?) then most users running this container. When I start the container it simply won't be able to access any of the files not created by the container itself.

**Comment by devedse** [comment]
Ok I figured out part of the issue here. Apparently you're making use of the LinuxServer images (thanks for sending me the Dockercompose file). However because you're overwriting the entrypoint, the default `/init` entrypoint is never fired and thus the logic to update the user is never kicked off.

So that would be the first thing that needs fixing, making sure the entry point remains the same, after that the dotnet process also needs to be ran as user `abc`. I'm trying to see if I can get an easy fix for this.

**Comment by devedse** [comment]
I figured out what the issue was. The AccessDenied errors threw me off and pointed me in the wrong direction.

In the end everything worked without setting the PGID and PUID except that the issue was that the media folder I mounted was mounted as Readonly.

In FileFlows a check is being done that checks for read AND write access to a file. It seems `using (var fs = File.Open(file.FullName, FileMode.Open))` is throwing an exception rather then nicely saying that write access is not allowed.

There's 2 ways to fix this:
1: Don't mount the directory as RO (readonly)
2: Disable Skip File Access Tests on the library:
![image](https://user-images.githubusercontent.com/2350015/203042222-d013b820-b683-4daa-bb06-cd66fefb263a.png)

A good improvement (in my oppinion) would be to elaborate a bit further in the logs on the `Access Denied` error that pops up when an exception is thrown.
